src/test/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/LaboratoryInvestigationResultTest.java

```diff
@@ -6,70 +6,126 @@
+            LaboratoryInvestigationResult.fromString("RSL+NV+11.9++:::ng/mL");
+
+        assertAll(
+            () -> assertThat(laboratoryInvestigationNumericalResult.getMeasurementValue())
+                .isEqualTo(new BigDecimal("11.9")),
+            () -> assertThat(laboratoryInvestigationNumericalResult.getMeasurementValueComparator()).isEmpty(),
+            () -> assertThat(laboratoryInvestigationNumericalResult.getMeasurementUnit()).isEqualTo("ng/mL"),
+            () -> assertThat(laboratoryInvestigationNumericalResult.getDeviatingResultIndicator()).isNull()
         );
+    }
+
+    @Test
+    void when_edifactString4IsPassed_expect_returnALaboratoryInvestigationCodedResultObject() {
+        final LaboratoryInvestigationResult laboratoryInvestigationCodedResult =
+            LaboratoryInvestigationResult.fromString("RSL+CV+::111222333:921::Cancer screening");
 
-        assertThat(laboratoryInvestigationResult)
-            .usingRecursiveComparison()
-            .isEqualTo(LaboratoryInvestigationResult.fromString("RSL+NV+11.9++:::ng/mL"));
+        assertAll(
+            () -> assertThat(laboratoryInvestigationCodedResult.getCode()).isEqualTo("111222333"),
+            () -> assertThat(laboratoryInvestigationCodedResult.getCodeType()).isEqualTo(CodingType.SNOMED_CT_CODE),
+            () -> assertThat(laboratoryInvestigationCodedResult.getDescription()).isEqualTo("Cancer screening")
+        );
     }
 
     @Test
     void when_buildingSegmentObjectWithoutAnyFields_expect_nullPointerExceptionIsThrown() {
-        assertThrows(NullPointerException.class, () -> LaboratoryInvestigation.builder().build());
+        assertThatNullPointerException().isThrownBy(() -> LaboratoryInvestigation.builder().build());
     }
 
     @Test
     void testGetKey() {
-        assertEquals(laboratoryInvestigationResult.getKey(), "RSL");
+        final LaboratoryInvestigationResult laboratoryInvestigationResult =
+            LaboratoryInvestigationResult.builder()
+                .measurementValue(new BigDecimal("11.9"))
+                .measurementValueComparator(MeasurementValueComparator.LESS_THAN)
+                .measurementUnit("ng/mL")
+                .deviatingResultIndicator(DeviatingResultIndicator.ABOVE_HIGH_REFERENCE_LIMIT)
+                .build();
+
+        assertThat(laboratoryInvestigationResult.getKey()).isEqualTo("RSL");
     }
 
     @Test
-    void testValidate() {
-        assertDoesNotThrow(laboratoryInvestigationResult::validate);
-
-        LaboratoryInvestigationResult emptyMeasurementValue = new LaboratoryInvestigationResult(null,
-            MeasurementValueComparator.LESS_THAN, "ng/mL", DeviatingResultIndicator.ABOVE_HIGH_REFERENCE_LIMIT);
-        LaboratoryInvestigationResult emptyMeasurementUnit = new LaboratoryInvestigationResult(new BigDecimal("11.9"),
-            MeasurementValueComparator.LESS_THAN, null, DeviatingResultIndicator.ABOVE_HIGH_REFERENCE_LIMIT);
+    void testValidateNumericalResult() {
```

Comment:
This has 3 different tests in it; can we split them out into different test methods?